File: 백준/Bronze/4153. 직각삼각형/직각삼각형.py
while True:
    lenght_list = list(map(int, input().split()))
    lenght_list.sort()
    if sum(lenght_list) == 0:
        break
    elif lenght_list[2] ** 2 == lenght_list[0] ** 2 + lenght_list[1] ** 2:
        print("right")
    else:
        print("wrong")

# while True 무한 반복
#   why? 종료 조건이 나중에 결정됨
# 입력 값을 리스트로
#   why? 큰 값을 골라내기 위해(정렬) c^2 = a^2 + b^2
# .sort()로 정렬
# 리스트의 합이 0이면 종료
#   why? 입력 조건이 양의 정수, 즉 0 0 0 인 경우만 합이 0
# 정렬된 리스트 인덱싱으로 큰 값 골라서 공식 대입 후 판단

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            if (a == 0) {
                break;
            }
            int[] arr = {a, b, c};
            Arrays.sort(arr);

            int result = (int) (Math.pow(arr[2], 2) - Math.pow(arr[1], 2) - Math.pow(arr[0], 2));

            if (result == 0) {
                System.out.println("right");
            } else {
                System.out.println("wrong");
            }
        }
    }
}